python/pyspark/ml/torch/distributor.py

```
@@ -0,0 +1,297 @@
+        return True
+    if lowercase_val == "false":
+        return False
+    raise Exception(
+        "get_conf_boolean expected a boolean conf "
+        "value but found value of type {} "
+        "with value: {}".format(type(val), val)
+    )
+
+
+class Distributor:
+    """
+    The parent class for TorchDistributor. This class shouldn't be instantiated directly.
+    """
+
+    def __init__(
+        self,
+        num_processes: int = 1,
+        local_mode: bool = True,
+        use_gpu: bool = True,
+        spark: Optional[SparkSession] = None,
+    ):
+        self.num_processes = num_processes
+        self.local_mode = local_mode
+        self.use_gpu = use_gpu
+        if spark:
+            self.spark = spark
+        else:
+            self.spark = SparkSession.builder.getOrCreate()
+        self.sc = self.spark.sparkContext
+        self.num_tasks = self._get_num_tasks()
+        self.ssl_conf = None
+
+    def _get_num_tasks(self) -> int:
+        """
+        Returns the number of Spark tasks to use for distributed training
+
+        Returns
+        -------
+            The number of Spark tasks to use for distributed training
+        """
+
+        if self.use_gpu:
+            if not self.local_mode:
+                key = "spark.task.resource.gpu.amount"
+                task_gpu_amount = int(self.sc.getConf().get(key, "0"))
+                if task_gpu_amount < 1:
+                    raise RuntimeError(f"'{key}' was unset, so gpu usage is unavailable.")
+                # TODO(SPARK-41916): Address situation when spark.task.resource.gpu.amount > 1
+                return math.ceil(self.num_processes / task_gpu_amount)
+            else:
+                key = "spark.driver.resource.gpu.amount"
+                if "gpu" not in self.sc.resources:
+                    raise RuntimeError("GPUs were unable to be found on the driver.")
+                num_available_gpus = int(self.sc.getConf().get(key, "0"))
```

if `num_available_gpus` == 0, we should raise error saying GPU resources is not configured properly ?

---

Sure, will add that check.